src/systems/media-frames.js

```diff
@@ -0,0 +1,219 @@
+import { isTagged, setTag } from "../components/tags";
+import { MediaType } from "../utils/media-utils";
+import { TEXTURES_FLIP_Y } from "../loaders/HubsTextureLoader";
+
+const TARGET_ZONE_STATE = {
+  EMPTY: "empty",
+  HOVERING: "hovering",
+  FULL: "full"
+};
+
+function scaleForAspectFit(containerSize, itemSize) {
+  return Math.min(containerSize.x / itemSize.x, containerSize.y / itemSize.y);
+}
+
+const isCapturableByType = {
+  [MediaType.ALL]: function(el) {
+    return !!(el && NAF.utils.isMine(el) && el.components["media-loader"]);
+  },
+  [MediaType.ALL_2D]: function(el) {
+    return !!(
+      el &&
+      NAF.utils.isMine(el) &&
+      (el.components["media-image"] || el.components["media-video"] || el.components["media-pdf"])
+    );
+  },
+  [MediaType.MODEL]: function(el) {
+    return !!(el && NAF.utils.isMine(el) && el.components["gltf-model-plus"]);
+  },
+  [MediaType.IMAGE]: function(el) {
+    return !!(el && NAF.utils.isMine(el) && el.components["media-image"]);
+  },
+  [MediaType.VIDEO]: function(el) {
+    return !!(el && NAF.utils.isMine(el) && el.components["media-video"]);
+  },
+  [MediaType.PDF]: function(el) {
+    return !!(el && NAF.utils.isMine(el) && el.components["media-pdf"]);
+  }
+};
+
+function getCapturableObjectCollidingWithBody(physicsSystem, mediaType, bodyUUID) {
+  const collisions = physicsSystem.getCollisions(bodyUUID);
+  for (let i = 0; i < collisions.length; i++) {
+    const bodyData = physicsSystem.bodyUuidToData.get(collisions[i]);
+    const mediaObjectEl = bodyData && bodyData.object3D && bodyData.object3D.el;
+    if (isCapturableByType[mediaType](mediaObjectEl)) {
+      return mediaObjectEl.object3D;
+    }
+  }
+  return null;
+}
+
+const components = [];
+export class MediaFramesSystem {
+  constructor(physicsSystem, interactionSystem) {
+    this.physicsSystem = physicsSystem;
+    this.interactionSystem = interactionSystem;
+  }
+
+  tick() {
+    for (let i = 0; i < components.length; i++) {
+      const zone = components[i];
+
+      const bodyUUID = zone.el.components["body-helper"].uuid;
+      // TODO would be nice to fix the timing so we don't need to check this
+      if (!this.physicsSystem.bodyInitialized(bodyUUID)) continue;
+
+      const objectToCapture = getCapturableObjectCollidingWithBody(this.physicsSystem, zone.data.mediaType, bodyUUID);
+
+      const previewMesh = zone.el.getObject3D("preview");
+      const guideMesh = zone.el.getObject3D("guide");
+
+      guideMesh.visible = this.interactionSystem.isHoldingAnything(isCapturableByType[zone.data.mediaType]);
```

Comment:
If we add networked state, might be nice if non-empty zones showed up differently.
Also might want guides to show up when hovering on an object in a zone? (Maybe only when in freeze mode?)

Reply:
They currently do show a different color when full, though like you said its not networked.